File: typescript/src/code-smells/oop-abusers/alternative-classes-different-interfaces.ts
export class TextLogger {
  log(message: string): void {
    console.log(`[text] ${message}`)
  }
}

export class MessageWriter {
  write(entry: string): void {
    console.log(`[text] ${entry}`)
  }
}

export function useAltClasses(choice: 'logger' | 'writer', msg: string): void {
  if (choice === 'logger') {
    new TextLogger().log(msg)
  } else {
    new MessageWriter().write(msg)
  }
}

File: csharp/src/code-smells/oop-abusers/AlternativeClassesDifferentInterfaces.cs
namespace RefactoringAvanzado.CodeSmells.OopAbusers;

public class TextLogger
{
    public void Log(string message)
    {
        Console.WriteLine($"[text] {message}");
    }
}

public class MessageWriter
{
    public void Write(string entry)
    {
        Console.WriteLine($"[text] {entry}");
    }
}

public static class AlternativeClassesDifferentInterfaces
{
    public static void UseAltClasses(string choice, string msg)
    {
        if (choice == "logger")
        {
            new TextLogger().Log(msg);
        }
        else
        {
            new MessageWriter().Write(msg);
        }
    }
}
